### PR TITLE
Sc 1619

### PR DESCRIPTION
**Pull request by amitec006**

### CODE REVIEW

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/OTPActor.java [hunk 1]
```diff
@@ -122,11 +120,32 @@ private void verifyOTP(Request request) {
       ProjectCommonException.throwClientErrorException(ResponseCode.errorInvalidOTP);
     }
 
+    if (otpInRequest.equals(otpInDB)) {
+      otpService.deleteOtp(type, key);
+    } else {
+      int remainingCount = getRemainingAttemptedCount(otpDetails);
```
Comment (manzarul) [inline actors/common/src/main/java/org/sunbird/learner/actors/otp/OTPActor.java]: This complete else code can be moved to separate private method , as handleMisMatchOtp or some thing.

File: actors/common/src/test/java/org/sunbird/learner/actors/otp/OTPActorTest.java [hunk 2]
```diff
@@ -144,27 +149,51 @@ private void verifyOtpFailureTest(boolean isPhone, Response mockedCassandraRespo
...
     ProjectCommonException exception =
-        probe.expectMsgClass(duration("10 second"), ProjectCommonException.class);
+        probe.expectMsgClass(duration("100 second"), ProjectCommonException.class);
     Assert.assertTrue(
         ((ProjectCommonException) exception)
             .getCode()
             .equals(ResponseCode.errorInvalidOTP.getErrorCode()));
   }
 
+  private void verifyOtpFailureTest(boolean isPhone, Response mockedCassandraResponse) {
```
Comment (manzarul) [inline actors/common/src/test/java/org/sunbird/learner/actors/otp/OTPActorTest.java]: have a look into this method , it either it should be @Test or removed Assert.

File: actors/common/src/test/java/org/sunbird/learner/actors/otp/OTPActorTest.java [hunk 3]
```diff
@@ -89,7 +94,7 @@ public void testVerifyOtpFailureWithInvalidEmailOtp() {
   @Test
   public void testVerifyOtpFailureWithExpiredOtp() {
     Response mockedCassandraResponse = getMockCassandraRecordByIdFailureResponse();
-    verifyOtpFailureTest(false, mockedCassandraResponse);
+    vrifyOtpFailureWithExpiredOtp(false, mockedCassandraResponse);
```
Comment (anmol2302) [inline actors/common/src/test/java/org/sunbird/learner/actors/otp/OTPActorTest.java]: verify type

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/OTPActor.java [hunk 4]
```diff
@@ -122,11 +120,35 @@ private void verifyOTP(Request request) {
...
+      otpService.deleteOtp(type, key);
+    } else {
+      handleMismachOtp(type, key, otpDetails);
+    }
     Response response = new Response();
     response.put(JsonKey.RESPONSE, JsonKey.SUCCESS);
     sender().tell(response, self());
   }
 
+  private void handleMismachOtp(String type, String key, Map<String, Object> otpDetails) {
```
Comment (anmol2302) [inline actors/common/src/main/java/org/sunbird/learner/actors/otp/OTPActor.java]: handleMismachOtp typo to handleMismatchOtp

File: actors/common/src/test/java/org/sunbird/learner/actors/otp/OTPActorTest.java (hunk 2 above)
Comment (amit-tarento) [reply actors/common/src/test/java/org/sunbird/learner/actors/otp/OTPActorTest.java]: this method is getting called from test method

File: actors/common/src/test/java/org/sunbird/learner/actors/otp/OTPActorTest.java (hunk 3 above)
Comment (amit-tarento) [reply actors/common/src/test/java/org/sunbird/learner/actors/otp/OTPActorTest.java]: done

File: actors/common/src/main/java/org/sunbird/learner/actors/otp/OTPActor.java (hunk 4 above)
Comment (amit-tarento) [reply actors/common/src/main/java/org/sunbird/learner/actors/otp/OTPActor.java]: done

**Review by anmol2302 — APPROVED** [review]

**Pull request merged**